### DISCUSSION TITLE
Source of CMake-ish environment variables in terminal

### DISCUSSION THREAD

**Discussion by mattgodbolt** [microsoft/vscode-cmake-tools#3819] [Q&A] [answered]
Hi folks,

I have a project that carefully manages all its various CMake variables externally (mostly within the CMakeList.txt itself). I just moved to using vscode and I realise that _somehow_ when I run a terminal inside vscode, some weird environment variables are being pre-populated that interfere with my runs:

```
$ env | grep gcc
GCC_NM=/home/mgodbolt/dev/tradewinds/out/env/bin/x86_64-conda-linux-gnu-gcc-nm
GCC=/home/mgodbolt/dev/tradewinds/out/env/bin/x86_64-conda-linux-gnu-gcc
GCC_RANLIB=/home/mgodbolt/dev/tradewinds/out/env/bin/x86_64-conda-linux-gnu-gcc-ranlib
```

These are not set in a regular terminal, and _appear_ to be coming from something that's parseing my CMake? As I say, these settings then interfere with me manually running `cmake`-like commands in my terminal (so I have to run them in a non-vscode terminal). Without getting into the "why is that the case" I'd like to understand where these shell variables are coming from and disabling it if possible.

Cheers, Matt

**Comment by mattgodbolt** [comment]
It seems like if I untick "Inherit env" in terminal, that "fixes" it. But I'd love to know what's going on here, if anyone has a clue.

**Reply by gcampbell-msft** [reply, marked as answer]
The "Inherit env" is coming from a different extension, so we don't have much insight into this behavior. Additionally, to determine if this is something from our extension, are you launching the terminal from an entry point in our extension? Or is this simply in the terminal when you run a command? 

We don't edit the environment that VS Code overall is sourced with from our extension, so if the entry point to the terminal is not coming from our extension, then I don't think it's an issue from it. 

Let me know if any of that is helpful! THanks.

**Comment by mattgodbolt** [comment]
Thank you! Sorry I didn't reply before.
